src/mapeo-manager.js

```diff
@@ -112,6 +143,32 @@ export class MapeoManager {
     return this.#rpc
   }
 
+  /**
+   * @param {'blobs' | 'icons'} mediaType
+   * @returns
+   */
+  async #getMediaBaseUrl(mediaType) {
+    await this.#deferredMediaServerListen.promise
+
+    let address = this.#mediaServer.server.address()
```

Comment:
Here I would use something like we had in [`getPort()`](https://github.com/digidem/mapeo-core-next/blob/1f4abac31b84fcc5a73a9db2ef7f8fafbffeb3ba/src/blob-server/index.js#L31-L40), probably wrapped in a [p-timeout](https://github.com/sindresorhus/p-timeout). We're going to be starting and stopping the media server whenever the app goes into the background, so the deferred promise will not work.